set.mm

```diff
@@ -500881,6 +501256,228 @@ The sumset operation can be used for both group (additive) operations and
   $}
 
 
+  ${
+    $d B x y $.  $d I x y $.  $d O x y $.  $d R x y $.
+    oppreqg.o $e |- O = ( oppR ` R ) $.
+    ${
+      oppreqg.b $e |- B = ( Base ` R ) $.
+      $( Group coset equivalence relation for the opposite ring.  (Contributed
+         by Thierry Arnoux, 9-Mar-2025.) $)
+      oppreqg $p |- ( ( R e. V /\ I C_ B ) -> ( R ~QG I ) = ( O ~QG I ) ) $=
+        ( vx vy wcel wss wa cqg co cv cpr cfv eqid eqgfval cvv copab wceq coppr
+        cminusg cplusg fvexi opprbas opprneg oppradd mpan adantl eqtr4d ) BEJZC
+        AKZLBCMNZHOZIOZPAKUPBUDQZQUQBUEQZNCJLHIUAZDCMNZHIUSUOCBUREAGURRZUSRZUOR
+        SUNVAUTUBZUMDTJUNVDDBUCFUFHIUSVACDURTAABDFGUGBURDFVBUHUSBDFVCUIVARSUJUK
+        UL $.
+    $}
+
+    $d O a b i x $.  $d R a b i x $.  $d a b i ph x $.
+    oppr2idl.2 $e |- ( ph -> R e. Ring ) $.
+    $( The ideals of the opposite ring's opposite ring are the ideals of the
+       original ring.  (Contributed by Thierry Arnoux, 9-Mar-2025.) $)
```

Comment:
The opposite ring's opposite ring is the original ring, isnt it?`This seems to be not proven yet. Maybe this is contained in your proof, an should be extracted.

Reply:
Yes, but! An opposite ring is a `sSet`, so [`setsidvald`](https://us.metamath.org/mpeuni/setsidvald.html) is the applicable theorem. Its hypotheses are things like `Fun R` and ``( .r ` ndx ) e. dom R``, and for [`tpostpos2`](https://us.metamath.org/mpeuni/tpostpos2.html), we need `Rel .x. /\ Rel dom .x.`, and the usual `R e. Ring` will not be sufficient.

I made that attempt, but decided not to post this proof.

Reply:
Here is my attempt: [opprabs.mmp](https://gist.github.com/tirix/c3086f22e46ad2681f13460c73720566)

Reply:
OK, I see , it is not that easy as it would be expected. Nevertheless, I would like to have ~opprabs in set.mm to visualize these difficulties.

Reply:
Done in https://github.com/metamath/set.mm/pull/4702/commits/9d63a4d7a9982b3829be673dac2a3babfa94ed23
I included a comment on the difficulties to use the theorem.